discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -16,47 +16,50 @@
 from netaddr import IPNetwork, IPAddress
 from test_infra import consts, utils
 from test_infra.controllers.load_balancer_controller import LoadBalancerController
+from test_infra.helper_classes.config import BaseClusterConfig
+from test_infra.helper_classes.nodes import Nodes
 from test_infra.tools import static_network, terraform_utils
-from tests.conftest import env_variables
 
 
 class Cluster:
 
-    def __init__(self, api_client, cluster_name=None, additional_ntp_source=None,
-                 openshift_version="4.7", cluster_id=None, user_managed_networking=False,
-                 high_availability_mode=consts.HighAvailabilityMode.FULL, olm_operators=None):
+    def __init__(self, api_client, config: BaseClusterConfig):
+        self._config = config
```

Comment:
make it public please, there is no need for it as private.  It should be easy to change any param from config

Reply:
Don't think the whole config needs to be public. If there is something we want to change or get we can define a setter/getter accordingly